src/sentry/snuba/metrics/helpers.py

```diff
@@ -449,11 +456,21 @@ def _build_queries(self, query_definition):
             for entity, fields in queries_by_entity.items()
         }
 
+    @staticmethod
+    def _build_conditional_aggregate_for_metric(entity, op, name):
+        snuba_function = OP_TO_SNUBA_FUNCTION[entity][op]
+        return Function(
+            snuba_function,
+            [Column("value"), Function("equals", [Column("metric_id"), resolve_weak(name)])],
+            alias=f"{op}({name})",
+        )
+
     @staticmethod
     def _build_select(entity, fields):
-        for op, _ in fields:
-            snuba_function = _OP_TO_SNUBA_FUNCTION[entity][op]
-            yield Function(snuba_function, [Column("value")], alias=op)
+        for op, name in fields:
+            yield SnubaQueryBuilder._build_conditional_aggregate_for_metric(
```

Comment:
nit: Let's make this a `@classmethod`, so we can do `cls._build_conditional_aggregate_for_metric`?